test/Tests/src/Data/Text_Spec.enso

```diff
@@ -256,6 +256,7 @@ spec =
             "Cześć" . contains 's\u{301}' Regex_Matcher.new . should_be_true
             'Czes\u{301}c\u{301}' . contains 's\u{301}' Regex_Matcher.new . should_be_true
             'Czes\u{301}c\u{301}' . contains 'ś' Regex_Matcher.new . should_be_true
+            'Czes\u{301}c\u{301}' . contains 's' Regex_Matcher.new . should_be_false
```

Comment:
Unfortunately, this test fails...

So Regex only works well with Unicode normalization to some extent - it does correctly find `ś` in `s\u{301}` and vice-versa. It does correctly **not** find `s` in `ś`. But it **incorrectly** (according to what I'd expect) **finds** `s` in `s\u{301}`. This is quite inconsistent. Maybe it actually should be reported as a bug in the Regex implementation - we already got one bug accepted there, so maybe we could get there with this one too - not exactly sure if this will be considered a bug, but the behaviour is not consistent - I don't think the results should depend on whether the string is normalized or not.

Not sure if simple workarounds exist for this - we could normalize the text before passing it to the engine but normalization _does_ split `ś` into `s\u{301}` (IIRC), so that would make it even worse (but at least consistent, irrelevant of if the input was normalized).

Reply:
Turns out Swift has a similar problem - actually it handles these cases even worse than our Regex
![image](https://user-images.githubusercontent.com/1436948/154953489-f1c1d742-259a-4d1c-aee2-6b214f85e246.png)

Seems to be a widely-known issue with Regex implementations: https://www.regular-expressions.info/unicode.html with no known implementations which do better in this case.

Will document this nuance in `contains` docstring and add unit tests showing it so that we are aware of it, but I expect we can't do much more than that.